### ISSUE TITLE
IE 11 only outputs inline styles

### ISSUE THREAD

**Issue by blackjacques** [mental-party/mejsx#1]
The results only show the inline style info for the <p id="lorem" style="line-height: 20px;"> element in IE 11, instead of:

//Order: 0 | Media:all
p { color: rgb(204, 204, 204); }

//Order: 1 | Media:all
#lorem { font-weight: bold; }

//Order: 2 | Media:(min-width: 300px), (min-device-width: 350px)
p { margin: 0px; }

//Order: 3 | Media:(min-width: 300px), (min-device-width: 350px)
#lorem { border: 1px solid rgb(0, 0, 0); }

//Order: 4 | Media:all
style {line-height: 20px;}

**Comment by blackjacques** [comment]
Changing the following lines seems to fix it:

var isCssMediaRule = function(cssRule) {
      return cssRule.type === cssRule.MEDIA_RULE; //cssRule.getName() === 'CSSMediaRule';
    }

    var isCssStyleRule = function(cssRule) {
      return cssRule.type === cssRule.STYLE_RULE; //cssRule.getName() === 'CSSStyleRule';
    }

**Comment by blackjacques** [comment]
The following function could be used to narrow down media queries to those that apply to the current document.  For instance, Media:(min-width: 300px) might be true on a desktop while Media:(max-width: 300px) would not.

    var mediaRuleMatchesDocument = function(cssRule) {
      return window.matchMedia( cssRule.media.mediaText ).matches;
    }

**Comment by er-han** [comment]
@blackjacques Thank you for your contribution.
I have added your fixing suggest and it did work on IE11.
But it didn't work on Edge so I did a bit change in the code, now it works on both Edge and IE11.

I am moving your latest function `mediaRuleMatchesDocument ` to a new issue, as it is not related to the bug you have found.

I am closing this issue now, if you see any misbehavior relevant to this bug we can open the issue again.

Thanks again.
